app-frontend/src/app/core/services/map.service.js

```diff
@@ -379,6 +379,16 @@ class MapWrapper {
         this.heldState = state;
         return this;
     }
+
+    /** Get the extent from a project (if it exists) and fit the map to it
+      * @param {Project} project to hold, e.g. specifying latlng center and zoom
+      * @returns {this} this
+      */
+    fitProjectExtent(project) {
```

Comment:
I think that's a good idea. I had debated internally over whether or not to put the function there.